src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/server/ProcessStreamConnectionProvider.java

```diff
@@ -160,5 +170,42 @@ public String toString() {
         return "ProcessStreamConnectionProvider [commands=" + this.getCommands() + ", workingDir=" //$NON-NLS-1$//$NON-NLS-2$
                 + this.getWorkingDirectory() + "]"; //$NON-NLS-1$
     }
+    public void checkJavaHome(String javaHome, String serverType) {
+
+        if (javaHome == null) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.java-home.null",serverType);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return;
+        }
+
+        File javaHomeDir = new File(javaHome);
+        if (!javaHomeDir.exists()) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.javaHomeDir.exist",serverType,Constants.REQUIRED_JAVA_VERSION,javaHome);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return;
+        }
+
+        if (!checkJavaVersion(javaHome, Constants.REQUIRED_JAVA_VERSION)) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.java-version", serverType,Constants.REQUIRED_JAVA_VERSION,javaHome);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+        }
+    }
+
+    private void notifyError(String errMsg, Project project) {
+        Notification notif = new Notification(Constants.LIBERTY_DEV_DASHBOARD_ID, errMsg, NotificationType.WARNING)
+                .setTitle(LocalizedResourceUtil.getMessage("liberty.action.cannot.start"))
```

Comment:
Guessing this was copied from the Liberty actions. Please verify that the dialog displayed is appropriate for these messages about the Java runtime used for the language servers. In particular the title, that would currently display "Liberty action was not able to run" should be replaced.